src/autoscaler/api/cred_helper/rpc.go

```diff
@@ -0,0 +1,64 @@
+package cred_helper
+
+import (
+	"autoscaler/models"
+	"net/rpc"
+)
+
+// Here is an implementation that talks over RPC
+type CredentialsRPC struct {
+	client *rpc.Client
+}
+
+func (g *CredentialsRPC) Create(appId string, userProvidedCredential *models.Credential) (*models.Credential, error) {
+	var resp string
+	err := g.client.Call("Plugin.Create", new(interface{}), &resp)
+	if err != nil {
+		// You usually want your interfaces to return errors. If they don't,
+		// there isn't much other choice here.
+		panic(err)
+	}
+
+	return nil, nil
+}
+
+func (g *CredentialsRPC) Delete(appId string) error {
+	var resp string
+	err := g.client.Call("Plugin.Delete", new(interface{}), &resp)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (g *CredentialsRPC) Get(appId string) (*models.Credential, error) {
```

Comment:
In one of the refactoring steps I think we should add the credential cache into this method.
```
res, found := a.credentialCache.Get(appID)
	if found {
		// Credentials found in cache
		credentials := res.(*models.Credential)
		isValid = a.validateCredentials(username, credentials.Username, password, credentials.Password)
	}
```
This will simplify the basic auth code  and add it into the communication layer so its "transparent".